tests/rspec/lib/cluster.rb

```
@@ -141,6 +147,35 @@ def wait_til_ready
     end
   end
 
+  def wait_nodes_ready
+    puts 'Waiting for nodes become in ready state after an update'
+    from = Time.now
+    loop do
+      begin
+        nodes = KubeCTL.run_and_parse(@kubeconfig, 'get nodes')
+        nodes_ready = Array.new(@tfvars_file.node_count, false)
+        nodes['items'].each_with_index do |item, index|
+          item['status']['conditions'].each do |condition|
+            if condition['type'] == 'Ready' && condition['status'] == 'True'
+              nodes_ready[index] = true
+            end
+          end
+        end
+
+        break if nodes_ready.uniq.length == 1
+
+        puts 'One or more nodes are not ready yet or missing nodes. Waiting...'
+        puts "# of returned nodes #{nodes['items'].size}. Expected #{@tfvars_file.node_count}"
+        raise StandardError
+      rescue KubeCTL::KubeCTLCmdError, StandardError
```

`StandardError` can be a lot of different issues, not only your `raise StandardError` from before. Thereby if anything else fails here, debugging it is very difficult, as this rescue just swallows the error message.